# UCA Downtown Acoustics

This notebook is to analyze the balloon-pop impulse acoustic data from the UCA Downtown location.

Date: April 27, 2022

In [1]:
from acoustics import bands,room

#octave(first octave center frequency,last octave center frequency)

octave_bands = bands.octave(125,4000)
print(octave_bands)

[ 125.  250.  500. 1000. 2000. 4000.]


## Room Response - No Treatment

In [2]:
import numpy as np
#t60_impulse(file_name, bands, rt='t30'):  # pylint: disable=too-many-locals
"""
Reverberation time from a WAV impulse response.
:param file_name: name of the WAV file containing the impulse response.
:param bands: Octave or third bands as NumPy array.
:param rt: Reverberation time estimator. It accepts `'t30'`, `'t20'`, `'t10'` and `'edt'`.
:returns: Reverberation time :math:`T_{60}`
"""

t60_bands_run_0_a = room.t60_impulse('run-0-a.wav',octave_bands,rt='t30')
t60_bands_run_0_b = room.t60_impulse('run-0-b.wav',octave_bands,rt='t30')
t60_bands_run_0_c = room.t60_impulse('run-0-c.wav',octave_bands,rt='t30')
print(t60_bands_run_0_a)
print(t60_bands_run_0_b)
print(t60_bands_run_0_c)

t60_bands_run_0_ave = np.zeros(len(octave_bands))
for i in range(len(octave_bands)):
    ave = (t60_bands_run_0_a[i] + t60_bands_run_0_b[i] + t60_bands_run_0_c[i]) / 3
    t60_bands_run_0_ave[i] = round(ave,1)

print('average reverb times =', t60_bands_run_0_ave)

[4.49811804 1.3576484  1.6068986  1.88542897 1.87518828 1.66727251]
[3.50479637 1.26177909 1.63846853 1.85619852 1.8096282  1.65521616]
[3.98604596 1.29484348 1.60056874 1.87847571 1.8930563  1.67819425]
average reverb times = [4.  1.3 1.6 1.9 1.9 1.7]


## 10 NRC 0.55 ceiling tiles on floor

In [3]:
#t60_impulse(file_name, bands, rt='t30'):  # pylint: disable=too-many-locals
"""
Reverberation time from a WAV impulse response.
:param file_name: name of the WAV file containing the impulse response.
:param bands: Octave or third bands as NumPy array.
:param rt: Reverberation time estimator. It accepts `'t30'`, `'t20'`, `'t10'` and `'edt'`.
:returns: Reverberation time :math:`T_{60}`
"""

t60_bands_run_1_a = room.t60_impulse('run-1-a.wav',octave_bands,rt='t30')
t60_bands_run_1_b = room.t60_impulse('run-1-b.wav',octave_bands,rt='t30')
t60_bands_run_1_c = room.t60_impulse('run-1-c.wav',octave_bands,rt='t30')
print(t60_bands_run_1_a)
print(t60_bands_run_1_b)
print(t60_bands_run_1_c)

t60_bands_run_1_ave = np.zeros(len(octave_bands))
for i in range(len(octave_bands)):
    ave = (t60_bands_run_1_a[i] + t60_bands_run_1_b[i] + t60_bands_run_1_c[i]) / 3
    t60_bands_run_1_ave[i] = round(ave,1)

print('average reverb times =', t60_bands_run_1_ave)

[4.10482292 1.28543249 1.55735597 1.81076112 1.74865959 1.55687351]
[4.15185369 1.18832145 1.54056238 1.71711079 1.7471239  1.55074093]
[3.2809162  1.21485375 1.51375056 1.69272835 1.73292926 1.58042441]
average reverb times = [3.8 1.2 1.5 1.7 1.7 1.6]


## 10 NRC 0.55 ceiling tiles plus 12 NRC 0.9 square ceiling tiles on floor

In [4]:
#t60_impulse(file_name, bands, rt='t30'):  # pylint: disable=too-many-locals
"""
Reverberation time from a WAV impulse response.
:param file_name: name of the WAV file containing the impulse response.
:param bands: Octave or third bands as NumPy array.
:param rt: Reverberation time estimator. It accepts `'t30'`, `'t20'`, `'t10'` and `'edt'`.
:returns: Reverberation time :math:`T_{60}`
"""

t60_bands_run_3_a = room.t60_impulse('run-3-a.wav',octave_bands,rt='t30')
t60_bands_run_3_b = room.t60_impulse('run-3-b.wav',octave_bands,rt='t30')
t60_bands_run_3_c = room.t60_impulse('run-3-c.wav',octave_bands,rt='t30')
print(t60_bands_run_3_a)
print(t60_bands_run_3_b)
print(t60_bands_run_3_c)

t60_bands_run_3_ave = np.zeros(len(octave_bands))
for i in range(len(octave_bands)):
    ave = (t60_bands_run_3_a[i] + t60_bands_run_3_b[i] + t60_bands_run_3_c[i]) / 3
    t60_bands_run_3_ave[i] = round(ave,1)

print('average reverb times =', t60_bands_run_3_ave)

[4.1866466  1.25377897 1.45500401 1.71160154 1.71088962 1.55369439]
[3.81015906 1.2959701  1.50444693 1.72407009 1.72022856 1.54705029]
[3.24218568 1.20919888 1.47365401 1.71648447 1.64920494 1.52293793]
average reverb times = [3.7 1.3 1.5 1.7 1.7 1.5]


## same as run 3 plus three moving blankets on floor

In [5]:
#t60_impulse(file_name, bands, rt='t30'):  # pylint: disable=too-many-locals
"""
Reverberation time from a WAV impulse response.
:param file_name: name of the WAV file containing the impulse response.
:param bands: Octave or third bands as NumPy array.
:param rt: Reverberation time estimator. It accepts `'t30'`, `'t20'`, `'t10'` and `'edt'`.
:returns: Reverberation time :math:`T_{60}`
"""

t60_bands_run_4_a = room.t60_impulse('run-4-a.wav',octave_bands,rt='t30')
t60_bands_run_4_b = room.t60_impulse('run-4-b.wav',octave_bands,rt='t30')
t60_bands_run_4_c = room.t60_impulse('run-4-c.wav',octave_bands,rt='t30')
print(t60_bands_run_4_a)
print(t60_bands_run_4_b)
print(t60_bands_run_4_c)

t60_bands_run_4_ave = np.zeros(len(octave_bands))
for i in range(len(octave_bands)):
    ave = (t60_bands_run_4_a[i] + t60_bands_run_4_b[i] + t60_bands_run_4_c[i]) / 3
    t60_bands_run_4_ave[i] = round(ave,1)

print('average reverb times =', t60_bands_run_4_ave)

[3.13556672 1.39437995 1.44927453 1.58315727 1.52897691 1.39941277]
[3.18932115 1.37192853 1.44118988 1.61716727 1.50497691 1.37208537]
[6.20427738 1.26919916 1.41982212 1.58061033 1.57097132 1.37350186]
average reverb times = [4.2 1.3 1.4 1.6 1.5 1.4]


## Summary of Results

|Run #|125 Hz|250 Hz|500 Hz|1000 Hz|2000 Hz|4000 Hz|
|:------:|:------:|:------:|:------:|:------:|:------:|:------:|
|run 0|4.0|1.3|1.6|1.9|1.9|1.7|
|run 1|3.8|1.2|1.5|1.7|1.7|1.6|
|run 3|3.7|1.3|1.5|1.7|1.7|1.5|
|run 4|4.2|1.3|1.4|1.6|1.5|1.4|

**run 0:** No acoustic treatment<br>
**run 1:** 10 NRC 0.55 ceiling tiles on floor<br>
**run 3:** 10 NRC 0.55 ceiling tiles plus 12 NRC 0.9 square ceiling tiles on floor<br>
**run 4:** 10 NRC 0.55 ceiling tiles plus 12 NRC 0.9 square ceiling tiles on floor plus 3 moving blankets on the floor<br>

# Run 6

In [2]:
import numpy as np
#t60_impulse(file_name, bands, rt='t30'):  # pylint: disable=too-many-locals
"""
Reverberation time from a WAV impulse response.
:param file_name: name of the WAV file containing the impulse response.
:param bands: Octave or third bands as NumPy array.
:param rt: Reverberation time estimator. It accepts `'t30'`, `'t20'`, `'t10'` and `'edt'`.
:returns: Reverberation time :math:`T_{60}`
"""

t60_bands_run_6_a = room.t60_impulse('run-6-a.wav',octave_bands,rt='t30')
t60_bands_run_6_b = room.t60_impulse('run-6-b.wav',octave_bands,rt='t30')
t60_bands_run_6_c = room.t60_impulse('run-6-c.wav',octave_bands,rt='t30')
print(t60_bands_run_6_a)
print(t60_bands_run_6_b)
print(t60_bands_run_6_c)

t60_bands_run_6_ave = np.zeros(len(octave_bands))
for i in range(len(octave_bands)):
    ave = (t60_bands_run_6_a[i] + t60_bands_run_6_b[i] + t60_bands_run_6_c[i]) / 3
    t60_bands_run_6_ave[i] = round(ave,1)

print('average reverb times =', t60_bands_run_6_ave)

[3.28622069 1.50407016 1.45724447 1.63689338 1.66893619 1.47707198]
[6.78683804 1.39776565 1.46995147 1.69673664 1.70098488 1.50592605]
[6.81711025 1.91702728 1.4862197  1.68454309 1.65059265 1.50590033]
average reverb times = [5.6 1.6 1.5 1.7 1.7 1.5]
